pkg/kubelet/cm/dra/manager_test.go

```diff
@@ -0,0 +1,405 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dra
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+	v1 "k8s.io/api/core/v1"
+	resourcev1alpha2 "k8s.io/api/resource/v1alpha2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/dynamic-resource-allocation/resourceclaim"
+	drapbv1 "k8s.io/kubelet/pkg/apis/dra/v1alpha2"
+	"k8s.io/kubernetes/pkg/kubelet/cm/dra/plugin"
+	"k8s.io/kubernetes/pkg/kubelet/cm/dra/state"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+)
+
+const (
+	driverName      = "test-cdi-device"
+	driverClassName = "test"
+)
+
+var (
+	socketName string
```

Comment:
Yes it is global to that package but I need to find a way to get rid off it because it is the endpoint used when registering the plugin and it is the result of:
```go
	socketDir, err := os.MkdirTemp("", "dra")
	if err != nil {
		t.Fatal(err)
	}

	socketName = filepath.Join(socketDir, "server.sock")
```
It produces something like: `/var/folders/zt/9d312jzj1pg5lywpbg_x77tc0000gp/T/dra2194543874/server.sock` on my laptop

Reply:
done in 04cfefa673c